### PR TITLE
Add endpoint for searching for users on a map

### PR DESCRIPTION
**Pull request by maurizi**
Intended for implementing OpenTreeMap/otm-core#388

Connects to #2864

### CODE REVIEW

File: opentreemap/treemap/views/user.py [hunk 1]
```diff
@@ -208,3 +208,34 @@ def user(request, username):
...
+                           .values('user_id', 'user__username',
+                                   'user__first_name', 'user__last_name',
+                                   'user__make_info_public')
+
+    if query:
+        users_qs = users_qs.filter(user__username__icontains=query)
+
+    if max_items:
+        users_qs = users_qs[:int(max_items)]
+
```
Comment (RobinIsTheBird) [inline opentreemap/treemap/views/user.py]: Whether or not to accept this depends on how much we expect it to get used on maps with tens of thousands of users. Maybe we need input from @dboyer.

If there are many results, but the search string is late in the alphabet, the matches that start with the search string (and therefore are best matches) are likely to be clipped from the result by `max_items`.

Having used many search boxes in other web apps that do exactly that, I can attest that it is annoying. But if we regard this feature as fringe, maybe we don't care.

@dboyer, do we care?

File: opentreemap/treemap/tests/test_views.py [hunk 2]
```diff
@@ -1955,3 +1955,45 @@ def test_instance_list_only_public(self):
...
+            self.assertIn('username', user_dict)
+            self.assertEqual(user.pk, user_dict['id'])
+            self.assertEqual(user.username, user_dict['username'])
+
+    def test_full_results(self):
+        users_list = users(make_request(), self.i2)
+        self.assert_users_in_list(users_list, self.matt)
+
+    def test_filtering(self):
+        users_list = users(make_request(), self.i1)
```
Comment (RobinIsTheBird) [inline opentreemap/treemap/tests/test_views.py]: Seems to me that using this request in `test_full_results` would spare having to repeat it here and in `test_max`.

**Comment by RobinIsTheBird** [comment]
:+1:

**Comment by maurizi** [comment]
@RobinIsTheBird I've improved the sorting of results, can you take another look?

**Comment by RobinIsTheBird** [comment]
:+1:

**Pull request merged**